File: src/main/java/no/nav/pensjon/regler/domain/beregning2011/BarnetilleggSerkullsbarnUT.java
package no.nav.pensjon.regler.domain.beregning2011;

import no.nav.pensjon.regler.domain.kode.FormelKodeCti;
import no.nav.pensjon.regler.domain.kode.YtelsekomponentTypeCti;

import java.io.Serializable;

/**
 * @author Ingleiv Johansen (Accenture) - PK-7250
 * @author Lars Hartvigsen (Decisive) - PK-20946
 */
public class BarnetilleggSerkullsbarnUT extends AbstraktBarnetilleggUT implements Serializable {

    private static final long serialVersionUID = -2197770710503560569L;
	
	/**
     * Brukers gjenlevendetillegg før justering.
     */
	// @GuiPrompt(prompt = "Brukers gjenlevendetillegg")
    private int brukersGjenlevendetilleggForJustering;

    public BarnetilleggSerkullsbarnUT() {
        super();
        ytelsekomponentType = new YtelsekomponentTypeCti("UT_TSB");
        setFormelKode(new FormelKodeCti("BTx"));
		
    }

    /**
     * Copy Constructor
     * 
     * @param barnetilleggFellesbarnUT an instance of {@link BarnetilleggSerkullsbarnUT} to copy values from
     */
    public BarnetilleggSerkullsbarnUT(BarnetilleggSerkullsbarnUT barnetilleggFellesbarnUT) {
        super(barnetilleggFellesbarnUT);
		brukersGjenlevendetilleggForJustering = barnetilleggFellesbarnUT.brukersGjenlevendetilleggForJustering;
		
    }
	
	public int getBrukersGjenlevendetilleggForJustering() {
		return brukersGjenlevendetilleggForJustering;
    }
	
	public void setBrukersGjenlevendetilleggForJustering(int brukersGjenlevendetilleggForJustering) {
       this.brukersGjenlevendetilleggForJustering = brukersGjenlevendetilleggForJustering;
    }
}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/BarnetilleggSerkullsbarnUT.kt
package no.nav.pensjon.regler.domain.beregning2011

import no.nav.pensjon.regler.domain.enum.FormelKodeEnum
import no.nav.pensjon.regler.domain.enum.YtelseskomponentTypeEnum
import java.io.Serializable

class BarnetilleggSerkullsbarnUT : AbstraktBarnetilleggUT, Serializable {
    /**
     * Brukers gjenlevendetillegg før justering.
     */
    var brukersGjenlevendetilleggForJustering = 0

    override val ytelsekomponentTypeEnum: YtelseskomponentTypeEnum = YtelseskomponentTypeEnum.UT_TSB

    constructor(barnetilleggFellesbarnUT: BarnetilleggSerkullsbarnUT) : super(barnetilleggFellesbarnUT) {
        brukersGjenlevendetilleggForJustering = barnetilleggFellesbarnUT.brukersGjenlevendetilleggForJustering
    }

    constructor() {
        formelKodeEnum = FormelKodeEnum.BTx
    }
}